compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -10,25 +10,216 @@ module Language.Bond.Codegen.Java.Class_java
+        typeName (BT_Maybe BT_UInt16) = [lt|com.microsoft.bond.StructBondType.SomethingUInt16StructField|]
+        typeName BT_UInt16 = [lt|com.microsoft.bond.StructBondType.UInt16StructField|]
+        typeName (BT_Maybe BT_UInt32) = [lt|com.microsoft.bond.StructBondType.SomethingUInt32StructField|]
+        typeName BT_UInt32 = [lt|com.microsoft.bond.StructBondType.UInt32StructField|]
+        typeName (BT_Maybe BT_UInt64) = [lt|com.microsoft.bond.StructBondType.SomethingUInt64StructField|]
+        typeName BT_UInt64 = [lt|com.microsoft.bond.StructBondType.UInt64StructField|]
+        typeName (BT_Maybe BT_Float) = [lt|com.microsoft.bond.StructBondType.SomethingFloatStructField|]
+        typeName BT_Float = [lt|com.microsoft.bond.StructBondType.FloatStructField|]
+        typeName (BT_Maybe BT_Double) = [lt|com.microsoft.bond.StructBondType.SomethingDoubleStructField|]
+        typeName BT_Double = [lt|com.microsoft.bond.StructBondType.DoubleStructField|]
+        typeName (BT_Maybe BT_Bool) = [lt|com.microsoft.bond.StructBondType.SomethingBoolStructField|]
+        typeName BT_Bool = [lt|com.microsoft.bond.StructBondType.BoolStructField|]
+        typeName (BT_Maybe BT_String) = [lt|com.microsoft.bond.StructBondType.SomethingStringStructField|]
+        typeName BT_String = [lt|com.microsoft.bond.StructBondType.StringStructField|]
+        typeName (BT_Maybe BT_WString) = [lt|com.microsoft.bond.StructBondType.SomethingWStringStructField|]
+        typeName BT_WString = [lt|com.microsoft.bond.StructBondType.WStringStructField|]
+        typeName (BT_Maybe (BT_UserDefined e@Enum {..} _)) = [lt|com.microsoft.bond.StructBondType.SomethingWStringStructField<#{qualifiedDeclaredTypeName java e}>|]
+        typeName(BT_UserDefined e@Enum {..} _) = [lt|com.microsoft.bond.StructBondType.EnumStructField<#{qualifiedDeclaredTypeName java e}>|]
+        typeName (BT_Maybe t) = [lt|com.microsoft.bond.StructBondType.SomethingObjectStructField<#{(getTypeName java) t}>|]
+        typeName t = [lt|com.microsoft.bond.StructBondType.ObjectStructField<#{(getTypeName java) t}>|]
+
+
+-- returns whether a struct field descriptor type (a protected nested class within StructBondType) is generic and hence needs an explicit type parameter
+isGenericStructFieldDescriptor :: Type -> Bool
+isGenericStructFieldDescriptor (BT_Maybe t) = not (isPrimitiveNonEnumBondType t)
+isGenericStructFieldDescriptor t = not (isPrimitiveNonEnumBondType t)
+
+
+-- returns an type descriptor expression for a given struct (identified by a name and generic parameters), used in intiialization of struct field descriptors
+structFieldDescriptorInitStructExpr :: MappingContext -> String -> [Type] -> Text
+structFieldDescriptorInitStructExpr java qualifiedTypeName params = javaStructFieldDescriptorInitStructExpr
+    where
+        javaStructFieldDescriptorInitStructExpr = [lt|#{typeCastExpr}getStructType(#{qualifiedTypeName}.class#{paramExprList params})|]
```

Comment:
Yes, I chose the style of resolving this for 'java' mapping context.

---
In reply to: [124938304](https://github.com/Microsoft/bond/pull/517#discussion_r124938304) [](ancestors = 124938304)